### ISSUE TITLE
"prefer-const" false positive

### ISSUE THREAD

**Issue by lll000111** [eslint/eslint#9988]
**Tell us about your environment**

* **ESLint Version:** 4.18.0
* **Node Version:** 9.5.0
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** none

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    'env': {
        'es6': true,
        'node': true
    }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
'use strict';

/*eslint prefer-const: "error"*/

// eslint: error 'eventInterval' is never reassigned. Use 'const' instead  prefer-const
let eventInterval;

const queueProcessorFn = () => {
    clearInterval(eventInterval);
};

eventInterval = setInterval(queueProcessorFn, 75);
```

```bash
eslint .
```

**What did you expect to happen?**

No errors. I cannot use `const eventInterval` in this code.

**What actually happened? Please include the actual, raw output from ESLint.**

```
$ eslint .

C:\Users\mha\OneDrive\Projects\Test\src\main.js
  6:5  error  'eventInterval' is never reassigned. Use 'const' instead  prefer-const

✖ 1 problem (1 error, 0 warnings)
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. However, this is working as intended -- you can use `const` like this:

```js
'use strict';

/*eslint prefer-const: "error"*/

const queueProcessorFn = () => {
    clearInterval(eventInterval);
};

const eventInterval = setInterval(queueProcessorFn, 75);
```
